presto-iceberg/src/test/java/org/apache/iceberg/rest/IcebergRestCatalogServlet.java

```diff
@@ -0,0 +1,240 @@
+import java.util.Collections;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.lang.String.format;
+
+/**
+ * The RESTCatalogServlet provides a servlet implementation used in combination with a
+ * RESTCatalogAdaptor to proxy the REST Spec to any Catalog implementation.
+ */
+public class IcebergRestCatalogServlet
+        extends HttpServlet
+{
+    private static final Logger LOG = Logger.get(RESTCatalogServlet.class);
+
+    private final RESTCatalogAdapter restCatalogAdapter;
+    private final Map<String, String> responseHeaders =
+            ImmutableMap.of(HttpHeaders.CONTENT_TYPE, ContentType.APPLICATION_JSON.getMimeType());
+
+    public IcebergRestCatalogServlet(RESTCatalogAdapter restCatalogAdapter)
+    {
+        this.restCatalogAdapter = restCatalogAdapter;
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response)
+            throws IOException
+    {
+        execute(ServletRequestContext.from(request), response);
+    }
+
+    @Override
+    protected void doHead(HttpServletRequest request, HttpServletResponse response)
+            throws IOException
+    {
+        execute(ServletRequestContext.from(request), response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws IOException
+    {
+        execute(ServletRequestContext.from(request), response);
+    }
+
+    @Override
+    protected void doDelete(HttpServletRequest request, HttpServletResponse response)
+            throws IOException
+    {
+        execute(ServletRequestContext.from(request), response);
+    }
+
+    protected void execute(ServletRequestContext context, HttpServletResponse response)
+            throws IOException
+    {
+        response.setStatus(HttpServletResponse.SC_OK);
+        responseHeaders.forEach(response::setHeader);
+
+        if (context.error().isPresent()) {
+            response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+            RESTObjectMapper.mapper().writeValue(response.getWriter(), context.error().get());
+            return;
+        }
+
+        try {
+            Object responseBody =
+                    restCatalogAdapter.execute(
+                            context.method(),
+                            context.path(),
+                            context.queryParams(),
+                            context.body(),
+                            context.route().responseClass(),
+                            context.headers(),
+                            handle(response));
+
+            if (responseBody != null) {
+                RESTObjectMapper.mapper().writeValue(response.getWriter(), responseBody);
+            }
+        }
+        catch (RESTException e) {
+            LOG.error("Error processing REST request", e);
```

Comment:
```suggestion
            LOG.error(e, "Error processing REST request");
```

Reply:
In my latest commit, I've gone ahead and removed the trace for this portion (more details in another comment!)